src/scenes/TasksView/index.tsx

```diff
@@ -231,17 +233,39 @@ function TasksView({ isAgentView }: TaskViewProps) {
             </h5>
           </div>
         )}
+ 
+        {!isAgentView && (
         <div className="flex flex-wrap gap-4">
           {tasks.map((task) => {
             return (
+              <Link to="/tasks/details">
               <TaskCard
                 key={task.id}
                 task={task}
                 completed={!isAgentView && completedTaskIdList.includes(task.id)}
               />
+              </Link>
             );
-          })}
+          })}        
         </div>
+        )}
+
+        {isAgentView && profile?.role === Profile.Role.HuddleAgent && (
+        <div className="flex flex-wrap gap-4">
+          {tasks.map((task) => {
+            return (
+              <Link to="/agent/tasks/details">
+              <TaskCard
+                key={task.id}
+                task={task}
+                completed={!isAgentView && completedTaskIdList.includes(task.id)}
+              />
+              </Link>
+           );
+         })}
+       </div>
+        )}
+
```

Comment:
I see what you've tried to do here, to send them to the correct details page depending on their role, 
This is a pretty good working solution and there's nothing wrong with it, But we can do the same thing with less code, we can just keep one of these tasks.map functions and use the link,
```javascript
<Link to={`${window.location.pathname/details}`}>
```
to navigate them to the correct page